src/ck-perf/trace-summary.h

```diff
@@ -30,16 +30,93 @@ class BinEntry {
 #if defined(_WIN32) || CMK_MULTIPLE_DELETE
     void operator delete(void *, void *) { }
 #endif
-    BinEntry(): _time(0.), _idleTime(0.) {}
-    BinEntry(double t, double idleT): _time(t), _idleTime(idleT) {}
+    BinEntry(): _time(0.), _idleTime(0.),
+                 _msgSize(0), _msgCount(0),
+                 _recvSize(0), _recvCount(0),
+                 _extRecvSize(0), _extRecvCount(0)
+    {
+      _msgCountPerEP = CkVec<unsigned long>(_entryTable.size() + 10);
+      _msgSizePerEP = CkVec<unsigned long>(_entryTable.size() + 10);
+      _recvCountPerEP = CkVec<unsigned long>(_entryTable.size() + 10);
+      _recvSizePerEP = CkVec<unsigned long>(_entryTable.size() + 10);
+      _extRecvCountPerEP = CkVec<unsigned long>(_entryTable.size() + 10);
+      _extRecvSizePerEP = CkVec<unsigned long>(_entryTable.size() + 10);
+      for(int i = 0; i < _msgCountPerEP.size(); ++i) {
+        _msgCountPerEP[i] = _msgSizePerEP[i] =
+                _recvCountPerEP[i] = _recvSizePerEP[i] =
+                        _extRecvCountPerEP[i] = _extRecvCountPerEP[i] = 0;
+      }
+    }
+
+    BinEntry(double t, double idleT, unsigned long msgSize, unsigned long msgCount,
+             CkVec<unsigned long> msgSizePerEP, CkVec<unsigned long> msgCountPerEP,
+             unsigned long recvSize, unsigned long recvCount,
+             CkVec<unsigned long> recvSizePerEP, CkVec<unsigned long> recvCountPerEP,
+             unsigned long extRecvSize, unsigned long extRecvCount,
+             CkVec<unsigned long> extRecvSizePerEP, CkVec<unsigned long> extRecvCountPerEP
+             ): _time(t), _idleTime(idleT),
+                 _msgSize(msgSize), _msgCount(msgCount),
+                 _recvSize(recvSize), _recvCount(recvCount),
+                 _extRecvSize(extRecvSize), _extRecvCount(extRecvCount)
```

Comment:
All of the vectors can be added in this initializer list rather than being initialized in the body.

Reply:
These are not vectors values, but integer values of the total counts. These are not printed anywhere but are useful to debug if something does not match. But this part can be removed entirely without affecting the logs.